### ISSUE TITLE
Make server message for 401 and 403 more emphasized (and clarify it is from the server).

### ISSUE THREAD

**Issue by bsutton** [dart-lang/pub#3368]
We are implementing a private repo for dart and having some issues around controlling the messaging output by the pub command when we return a 401 or 403.

Our problem is that our system manages the pub tokens (pub add token) on behalf of the user so they never directly interact with the pub token commands.

When we return a 401 or a 403 the pub command prints our message (returned with the bearer header) but it also prints its own message wrt to tokens.

This is going to cause confusion with our users because they may not even know about pub tokens and they will see two conflicting messages.

As such we are looking for a method to suppress the standard pub message and just display the one we are returning.

We see two possible paths (and are open to others).

1) we add another parameter to the bearer such as 'override: true'
When pub sees the override it knows to suppress its own message and just display the message returned by the server.
2) we return an error other than 401/403 and pub displays that error.

On a side note the current 403 error doesn't actually make much sense. This is a forbidden error but when the client's identity is known to the server. In pubs case, this means that the token is valid (as it carries the identity) but the user tried to access content they don't have access to. As such I think the current message which advises how to add a token makes little sense and is even perhaps misleading. In our case we have a permission structure as to what packages a user has access to. So they can be authed but not allowed to get certain packages.

**Comment by jonasfj** [comment]
I don't think we should add more parameters to this messaging. But perhaps we can improve the message, I think we're open to that.

----------------

As I understand it the current messages are:

### 401
See [golden test](https://github.com/dart-lang/pub/blob/master/test/testdata/goldens/hosted/fail_gracefully_on_bad_version_listing_response_test/401-with-message.txt)

```
$ dart pub get
Resolving dependencies...
Because myapp depends on foo any which doesn't exist (authentication failed), version solving failed.

http://localhost:$PORT package repository requested authentication!
You can provide credentials using:
    pub token add http://localhost:$PORT
<message>
```


### 403
See [golden test](https://github.com/dart-lang/pub/blob/master/test/testdata/goldens/hosted/fail_gracefully_on_bad_version_listing_response_test/403-with-message.txt)

```
$ dart pub get
Resolving dependencies...
Because myapp depends on foo any which doesn't exist (authorization failed), version solving failed.

Insufficient permissions to the resource at the http://localhost:$PORT package repository.
You can modify credentials using:
    pub token add http://localhost:$PORT
<message>
```

-------------

Talking to @sigurdm, we think that it would be reasonable to adjust these messages, keeping in mind that we want to:
 * (a) Reference `dart pub token add <host>`, because in most cases this is what the user should be using.
 * (b) Highlight that the `<message>` originates from the server.

If you have a proposal for these message that does (a) and (b) better than what we have, I think we're open to considering it.

**Comment by bsutton** [comment]
@jonasfj @sigurdm 
Can I ask what the resistance is to adding more parameters to the messages?

Whilst most users 'may' need to use add token we are making presumptions about how each private package repo implementor will manage the auth.

For our users the process is:
```
dart pub global activate onepub
onepub login // triggers an oauth process 
cd mypackage
onepub private // updates publish_to
dart pub publish
```

The command onepub login essentially does the token add under the hood (we actually write directly to the pub-tokens.json file).

I very much doubt that we will be the only ones to offer this convenience to our users.

Adding the override parameter makes the messaging fully customisable so that 'all' users can have a great experience not just the many.

If it helps, we are prepared to submit a PR with the changes.

**Comment by sigurdm** [comment]
I still think pub should provide its instructions for the token commands regardless of what the server says. It is not that confusing.

We should probably still prefix and highlight the server message better.
